example/spec/parallel/discoverGranules/DiscoverGranulesDuplicateHandlingErrorSpec.js

```diff
@@ -0,0 +1,177 @@
+      prefix = config.stackName;
+      sourceBucket = config.bucket;
+
+      // The S3 path where granules will be ingested from
+      const sourcePath = `${prefix}/tmp/${randomId('test-')}`;
+
+      // Create the collection
+      collection = await createCollection(
+        prefix,
+        {
+          duplicateHandling: 'error',
+          provider_path: `${sourcePath}/`
+        }
+      );
+
+      // Create the S3 provider
+      provider = await createProvider(prefix, { host: sourceBucket });
+
+      // Stage the existing granule file to S3
+      existingGranuleId = randomId('existing-granule-');
+      existingGranuleKey = `${sourcePath}/${existingGranuleId}.txt`;
+      await s3PutObject({
+        Bucket: sourceBucket,
+        Key: existingGranuleKey,
+        Body: 'asdf'
+      });
+
+      // Ingest the existing granule
+      ingestGranuleRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'IngestGranule',
+          collection: pick(collection, ['name', 'version']),
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-'),
+            granules: [
+              {
+                granuleId: existingGranuleId,
+                dataType: collection.name,
+                version: collection.version,
+                files: [
+                  {
+                    name: `${existingGranuleId}.txt`,
+                    path: sourcePath
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      );
+
+      // Find the "IngestGranule" execution ARN
+      const ingestGranuleExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === ingestGranuleRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+
+      // Wait for the "IngestGranule" execution to be completed
+      await getCompletedExecution({ prefix, arn: ingestGranuleExecutionArn });
+
+      // Wait for the existing granule to be fully ingested
+      await getCompletedGranule({ prefix, granuleId: existingGranuleId });
+
+      // Run DiscoverGranules
+      discoverGranulesRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'DiscoverGranules',
+          collection: {
+            name: collection.name,
+            version: collection.version
+          },
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-')
+          }
+        }
+      );
+
+      // Find the "DiscoverGranules" execution ARN
+      discoverGranulesExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === discoverGranulesRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+    } catch (err) {
+      beforeAllFailed = true;
+      console.error(err);
+      throw err;
+    }
+  });
```

Comment:
There were a few motivations here:

1. The original test file was actually testing two separate things: duplicate handling set to `skip` and duplicate handling set to `error`. Breaking those apart into two separate tests made it easier to see what was being tested, makes the tests run a bit faster since they can now run in parallel, and, most importantly, prevents one test from interfering with another.
2. The original test was using hard-coded granule IDs, collections, and providers. Refactoring it like this allowed for randomized test data.
3. Jonathan and I talked it over and agreed that the integration tests should be more from the user's perspective than they are now. Operators wouldn't be querying and writing to the DB, and they wouldn't be starting workflows through the AWS API, they would be doing it through the Cumulus API. By writing the tests at that level, we have already uncovered a number of bugs and issues.

Reply:
4. The original test was using the `DiscoverGranules` workflow to populate the initial test granules into Cumulus. That eventually triggers `IngestGranule`, which actually creates the granules. It was more efficient to just invoke `IngestGranule` directly to stage that test data.